File: backend/internal/app/models/game.go
package models

type Game struct {
	ID         string   `json:"id" bson:"_id,omitempty"`
	Package    string   `json:"package" bson:"package"`
	Type       string   `json:"type" bson:"type"`
	CaseType   []string `json:"case_type" bson:"case_type"`
	GameEngine string   `json:"game_engine" bson:"game_engine"`
	GameUrl    string   `json:"game_url" bson:"game_url"`
	GameName   string   `json:"game_name" bson:"game_name"`
	GameType   string   `json:"game_type" bson:"game_type"`
	GameId     int      `json:"game_id" bson:"game_id"`
	Status     bool     `json:"status" bson:"status"`
}

type GameRequest struct {
	GameType string `json:"game_type" bson:"game_type"`
	CaseType string `json:"case_type" bson:"case_type"`
	GameName string `json:"game_name" bson:"game_name"`
}

File: web/src/api/game.ts
import request from "@/utils/request";


export interface Game {
  id: String;
  package: String;
  type: String;
  case_type: String[];
  game_engine: String;
  game_url: String;
  game_name: String;
  game_type: String;
  game_id: number;
  status: boolean;
}

export interface SearchGame {
  game_type: string;
  case_type: string;
  game_name: string;
}

export const findAllWxGame = () => {
  return request({
    url: '/api/v1/game/find_wx',
    method: 'get'
  });
}

export const findRpkGame = () => {
  return request({
    url: '/api/v1/game/find_rpk',
    method: 'get'
  })
}

export const searchGame = (data: SearchGame) => {
  return request({
    url: '/api/v1/game/find_by_type',
    method: 'post',
    data: data
});
}

export const addGame = (data: Game) => {
  return request({
    url: '/api/v1/game/add',
    method: 'post',
    data: data
  });
}

export const updateGame = (data: Game) => {
  return request({
    url: '/api/v1/game/update',
    method: 'put',
    data: data
  });
}

export const updateByFeishu = () => {
  return request({
    url: '/api/v1/game/feishu',
    method: 'get',
  })
}
